File: test_error_tracking.py
# Test error tracking across all SDKs

import sys
import os

# Ensure we can import toondb
sys.path.insert(0, os.path.join(os.path.dirname(__file__), 'toondb-python-sdk', 'src'))

from toondb.analytics import capture_error, is_analytics_disabled

print("=" * 60)
print("Testing Error Tracking (Static Only)")
print("=" * 60)
print(f"Analytics disabled: {is_analytics_disabled()}\n")

if not is_analytics_disabled():
    print("Sending test error events (static info only)...\n")
    
    # Test different error types with static locations
    errors = [
        ("connection_error", "database.open"),
        ("query_error", "sql.execute"),
        ("permission_error", "table.access"),
        ("timeout_error", "transaction.commit"),
    ]
    
    for i, (error_type, location) in enumerate(errors, 1):
        print(f"{i}. {error_type} @ {location}...")
        capture_error(error_type, location)
        print(f"   ✅ Sent")
    
    print("\n" + "=" * 60)
    print("All error events sent successfully!")
    print("Only static info sent - no sensitive data!")
    print("=" * 60)
else:
    print("⚠️  Analytics disabled - no events sent")

File: test_error_tracking.js
// Test error tracking in JavaScript/Node.js

const { captureError, isAnalyticsDisabled } = require('./sochdb-js/dist/cjs/index.js');

async function testErrorTracking() {
  console.log('='.repeat(60));
  console.log('Testing Error Tracking (Static Only)');
  console.log('='.repeat(60));
  console.log(`Analytics disabled: ${isAnalyticsDisabled()}\n`);

  if (!isAnalyticsDisabled()) {
    console.log('Sending test error events (static info only)...\n');

    const errors = [
      ['connection_error', 'database.open'],
      ['query_error', 'sql.execute'],
      ['permission_error', 'table.access'],
      ['timeout_error', 'transaction.commit'],
    ];

    for (let i = 0; i < errors.length; i++) {
      const [errorType, location] = errors[i];
      console.log(`${i + 1}. ${errorType} @ ${location}...`);
      await captureError(errorType, location);
      console.log(`   ✅ Sent`);
    }

    console.log('\n' + '='.repeat(60));
    console.log('All error events sent successfully!');
    console.log('Only static info sent - no sensitive data!');
    console.log('='.repeat(60));
  } else {
    console.log('⚠️  Analytics disabled - no events sent');
  }
}

testErrorTracking().catch(err => {
  console.error('Error:', err);
  process.exit(1);
});
